yocto-update-strategies.md

```diff
@@ -0,0 +1,651 @@
+  - ensuring the build environment is up-to-date and compatible with the Yocto
+    Project requirements
+* External layers update
+  - updating external layers used in the project to their latest versions
+* Custom layer update
+  - updating the custom layer(s) specific to the project, incorporating
+    necessary changes and improvements
+* Real-life examples
+  - responsive multimedia casting device using RaspberryPi
+  - Yocto BSP maintenance
+  - secure gateway for IoT data transfer using i.MX6
+  - full build automation in project using Rockchip RK3399
+
+???
+
+Now let's dive into the process of performing updates in Yocto-based projects.
+This stage is crucial for integrating new features, addressing security
+vulnerabilities, and ensuring the system remains up to date. We will discuss
+three key aspects to consider during the update process.
+
+First, the build environment needs to be reviewed and updated. It's important to
+ensure that the build environment is up-to-date and meets the requirements
+specified in the Yocto Project documentation. This ensures compatibility and a
+smooth update process.
+
+Next, we need to update the external layers used in the project. External layers
+provide additional functionality and components to the Yocto-based project. It's
+important to check for updates to these external layers and incorporate the
+latest versions to benefit from bug fixes, security patches, and new features.
+
+Finally, we have the custom layer update. Custom layers are specific to the
+project and may contain project-specific modifications, configurations, or
+recipes. During the update process, we need to review and update these custom
+layers to incorporate necessary changes, improvements, or bug fixes. This
+ensures that the project's customizations remain compatible and functional with
+the updated system.
+
+By addressing these three aspects—build environment, external layers, and custom
+layer update—we can ensure a smooth and successful update process for
+Yocto-based projects. Updating these components in a systematic manner reduces
+the risk of compatibility issues, ensures the project remains up to date, and
+allows for seamless integration of new functionalities and improvements.
+
+---
+# Performing update - build environment
+
+* Verify the compatibility of the build environment with the desired Yocto
+  Project release
+  - see `Build Host Packages` in Mega Manual
+  - use correct release
+* Prepare reproducible build environment
+  - ansible can be used to put together such an environment
+  - other possibility is to prepare containeraized image; e.g.
+    [kas](https://github.com/siemens/kas),
+    [imx-docker](https://github.com/nxp-imx/imx-docker),
+    [yocto-docker](https://github.com/3mdeb/yocto-docker)
+  - easier to distribute such an environment among colleagues
+* Review and update the configuration files (local.conf, bblayers.conf) to
+  reflect any changes or additions required for the update
```

Comment:
Maybe this can be bullet point under the `Review ...`?

Reply:
whole slide replaced with image https://github.com/3mdeb/conferences/pull/7/commits/5d224bb9561a6873714203312f58eea470e81930